### DISCUSSION TITLE
Syntax highlight in  ./layouts/index.html

### DISCUSSION THREAD

**Discussion by StefanSalewski** [thuliteio/doks#1037] [General]
In https://getdoks.org/docs/tutorial/add-pages/

we have

Customize the homepage

Open ./layouts/index.html in your code editor

I wonder if I can present some syntax highlighted source code on this page? Maybe replace

```
<p>Switch to a low-light UI with the click of a button. Change colors with variables to match your branding.</p>
```

with a small source code snipped shown with syntax highlight for a programming language like golang. I know that Hugo supports syntax highlight, but I got it not to work. Maybe it works only for separate markdown files?

References: https://gohugo.io/content-management/syntax-highlighting/

**Comment by StefanSalewski** [comment]
Actually, I think I do not need syntax highlight for ./layouts/index.html. But I am still interested if it is possible. I had already some trouble to get Monokai syntax highlight for Nim language working in markdown files at all, but now I finally got it. I hope I can remember tomorrow how I finally did it, then I will write some notes for it. It is really not that easy. It was also not easy to create a new menu, maybe called Books, as a replacement for Docs. One has to know, that we have to do "cp -r layouts/docs/ layouts/books/ to get it working.

**Comment by h-enk** [comment]
> Maybe it works only for separate markdown files?

Yes, that's correct. You could add `{{ .Content }}` to `./layouts/index.html` and add content to `./content/en/_index.md` e.g.

````bash
```bash
npm create hyas@latest
```
````

![Snag_11353c6a](https://user-images.githubusercontent.com/3902872/232227128-e424ed06-31bf-43fc-96e3-dff114c7f3a5.png)
